App.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+
+const App = () => {
+    const [todos, setTodos] = useState([]);
+    const [newTodo, setNewTodo] = useState('');
+
+    useEffect(() => {
+        fetch('http://localhost:3000/todos')
+            .then(response => response.json())
+            .then(data => setTodos(data));
+    }, []);
+
+    const addTodo = () => {
+        fetch('http://localhost:3000/todos', {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ title: newTodo }),
+        })
+            .then(response => response.json())
+            .then(data => setTodos([...todos, data]));
+
+        setNewTodo('');
+    };
+
+    const toggleComplete = id => {
+        const todo = todos.find(t => t.id === id);
+        fetch(`http://localhost:3000/todos/${id}`, {
+            method: 'PUT',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ completed: !todo.completed }),
+        })
+            .then(response => response.json())
+            .then(data => setTodos(todos.map(t => t.id === id ? data : t)));
+    };
+
+    return (
+        <div>
```

Comment:
Create separate component called itemList.js which returns <input /> and <ul></ul>   for code spliting.